services/user/AdminSys/ui/src/configuration/utils.ts

```diff
@@ -201,6 +201,57 @@ export function writeLoggers(loggers: { [index: string]: LogConfig }): {
     return result;
 }
 
+function readListen(listen: ListenConfig): ListenConfig {
+    const result = { ...listen };
+    delete result.address;
+    delete result.port;
+    delete result.path;
+    if (listen.protocol == "http" || listen.protocol == "https") {
+        if (listen.address == "0.0.0.0") {
+            return { text: String(listen.port), ...result };
+        } else if (listen.address?.includes(":")) {
+            return {
+                text: "[" + listen.address + "]:" + listen.port,
+                ...result,
+            };
+        } else {
+            return { text: listen.address + ":" + listen.port, ...result };
+        }
+    } else if (listen.protocol == "local") {
+        return { text: listen.path, ...result };
+    }
+    return listen;
+}
+
+function writeListen(listen: ListenConfig): any {
+    if (listen.protocol == "http" || listen.protocol == "https") {
+        const ipv4 = listen.text?.match(/^(\d+\.\d+.\d+.\d+)(?::(\d+))$/);
+        if (ipv4) {
+            return {
+                protocol: listen.protocol,
+                address: ipv4[1],
+                port: Number(ipv4[2]),
+            };
+        }
+        const ipv6 = listen.text?.match(/^\[(.*)\](?::(\d+))$/);
```

Comment:
The .* may be over-permissive

Reply:
It doesn't really matter here. This isn't being used for validation and it sufficient to distinguish the form of any valid inputs.